### ISSUE TITLE
[BUG] Auto-logged confusion matrix uses y_pred instead of y_true

### ISSUE THREAD

**Issue by gokhanceyhan** [mlflow/mlflow#4018]
### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [x] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS Catalina 10.15.7
- **MLflow installed from (source or binary)**: binary
- **MLflow version (run ``mlflow --version``)**: 1.13
- **Python version**: 3.6.9
- **npm version, if running the dev UI**:
- **Exact command to reproduce**: 
```
mlflow.sklearn.autolog()
```

### Describe the problem
When I turned on auto-logging for a classifier, the confusion matrix logged under the artifacts implies an accuracy score of 1 although the logged accuracy score is smaller than 1. I think the `y_true` argument in the following code should be set to `y_true` instead of `y_pred` (mlflow.sklearn.utils.py line 300):

```
if _is_plotting_supported():
        labels = set(y_true)
        classifier_artifacts.extend(
            [
                _SklearnArtifact(
                    name=_TRAINING_PREFIX + "confusion_matrix",
                    function=sklearn.metrics.plot_confusion_matrix,
                    arguments=dict(
                        estimator=fitted_estimator,
                        X=X,
                        y_true=y_pred,
                        sample_weight=sample_weight,
                        normalize="true",
                        cmap="Blues"
                    ),
                    title="Normalized confusion matrix",
                ),
            ]
        )
```

### Code to reproduce issue
```
from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split

import mlflow
import mlflow.sklearn


if __name__ == "__main__":
    
    X, y = make_classification()
    X_train, X_test, y_train, y_test = train_test_split(X, y)

    # enable auto-logging
    mlflow.sklearn.autolog()

    with mlflow.start_run():
        model = LogisticRegression()
        model.fit(X_train, y_train)
```

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by harupy** [comment]
#3993 has fixed this issue.

**Comment by harupy** [comment]
https://github.com/mlflow/mlflow/blob/9ca61a412f26ffede61a8fdf0b2921042b928eb4/mlflow/sklearn/utils.py#L307
